### PR TITLE
Fix resource issues with ciao-cli image

### PR DESCRIPTION
**Pull request by rbradford**
@markdryan found some resource issues with ciao-cli image commands in #662 this PR addresses those by not loading the entire image into memory when uploading and downloading.

### CODE REVIEW

**Comment by markdryan** [comment]
Looks good to me.  I was able to upload new images.  The upload was comparable too or perhaps slightly quicker than the old code.  I was going to suggest wrapping the file handle we're passing to images.Upload in a bufio.NewReader() but this doesn't work as images.Upload wants a ReadSeeker.  So looks good to me.

File: ciao-cli/image.go [hunk 1]
```diff
@@ -362,19 +361,19 @@ func (cmd *imageDownloadCommand) run(args []string) error {
...
-		fatalf("Could not read [%s]\n", err)
-	}
-
-	if cmd.file == "" {
-		fmt.Printf("%s\n", b)
-	} else {
-		err := ioutil.WriteFile(cmd.file, b, 0644)
+	dest := os.Stdout
+	if cmd.file != "" {
+		dest, err = os.Create(cmd.file)
```
Comment (markdryan) [inline ciao-cli/image.go]: We're not closing the file we're creating.  I think we need to close it and check the err from close to be sure all the data is correctly written.

Comment (rbradford) [inline ciao-cli/image.go]: Fixed with a defer!

Comment (markdryan) [inline ciao-cli/image.go]: we should really check the return value from close as we're writing to the file.

Comment (rbradford) [inline ciao-cli/image.go]: okay. done.

**Pull request merged**